src/main/scala/com/fulcrumgenomics/util/IntervalListSource.scala

```diff
@@ -0,0 +1,126 @@
+
+  /** Creates a new interval list source from a File. */
+  def apply(file: File): IntervalListSource = apply(path=file.toPath)
+
+  /** Creates a new interval list source from a Path. */
+  def apply(path: PathToIntervals): IntervalListSource = apply(Io.toInputStream(path))
+}
+
+/**
+  * Reads intervals from any text based source via a reader.  The underlying reader is closed automatically when EOF is
+  * reached.
+  */
+class IntervalListSource private(lines: Iterator[String],
+                                 private[this] val source: Option[{ def close(): Unit }] = None)
+  extends Iterator[Interval] with Closeable {
+
+  private val iter = lines.bufferBetter
+
+  private var lineNumber = 1L
+
+  // Read the header
+  val header: SAMFileHeader = {
+    val codec = new SAMTextHeaderCodec
+    val headerLines = iter.takeWhile(_.startsWith("@")).toSeq
+    require(headerLines.nonEmpty, "No header found")
+    lineNumber += headerLines.length
+    val lineReader = BufferedLineReader.fromString(headerLines.mkString("\n"))
+    codec.decode(lineReader, "IntervalListSource")
+  }
+
+  require(!this.dict.isEmpty, "No reference sequences found in the header.")
+
+  def dict: SAMSequenceDictionary= this.header.getSequenceDictionary
+
+  private val parseArray = Array[String]("", "", "", "", "")
+
+  /** True if calling `next()` will yield another record, false otherwise. */
+  override def hasNext: Boolean = this.iter.nonEmpty
+
+  /** Returns the next record if available, or throws an exception if none is available. */
+  override def next(): Interval = parse(this.iter.next())
+
+  /** Closes the underlying reader; only necessary if EOF hasn't been reached. */
+  override def close(): Unit = this.source.foreach(_.close())
+
+  private def parse(line: String): Interval = {
+    require(StringUtil.split(line, '\t', parseArray) == 5, s"Expected 5 fields on line $lineNumber")
+    val refName     = parseArray(0)
+    val startString = parseArray(1)
+    val endString   = parseArray(2)
+    val strand      = parseArray(3)
+    val name        = parseArray(4)
+
+    val start = startString.toInt
+    val end   = endString.toInt
+
+    Option(dict.getSequence(refName)) match {
+      case None => throw new IllegalArgumentException(f"Reference contig '$refName' not found in the sequence dictionary.")
+      case Some(seq) =>
+        require(1 <= start, s"Start is less than 1 on line number $lineNumber")
+        require(end <= seq.getSequenceLength, s"End is beyond the reference contig length on line number $lineNumber")
+        require(start <= end, f"Start is greater than end on line number $lineNumber")
+    }
+
+    val negative = strand match {
+      case "-" => true
+      case "+" => false
+      case _   => throw new IllegalArgumentException(s"Unrecognized strand '$strand' on line number $lineNumber")
+    }
+
+    lineNumber += 1
+
+    new Interval(refName, start, end, negative, name)
+  }
+}
```

Comment:
How about adding this so that we could use this as our default way to also load interval lists?

```scala
  def toIntervalList: IntervalList = {
    val list = new IntervalList(dict)
    while (hasNext) list.add(next())
    list
  }
```